src/lib/awsConnectionParameters.ts

```diff
@@ -102,18 +121,45 @@ function attemptEndpointCredentialConfiguration(
     if (!endpointName) {
         return undefined
     }
+    const authInfo = getEndpointAuthInfo(awsparams, endpointName)
+    authInfo.accessKey = authInfo.accessKey ?? ''
+    authInfo.secretKey = authInfo.secretKey ?? ''
```

Comment:
Since this change makes these two parameters (username/password) optional, it would be possible to accidentally leave one of those fields blank, when you aren't intending to assume a role. To help prevent these errors, `attemptEndpointCredentialConfiguration` should check if either of accessKey/secretKey are undefined, and if they are: log an error, and possibly return undefined (I'm not sure how the result is handled, so @rbbarad or @JadenSimon may have better guidance there).

Reply:
@awschristou : Those two lines are actually unchanged compared to the master branch. Previously, it was also possible both fields were blank, but I agree it was unlikely only one was blank. (It could happen if you set one via env vars).

I will add a check for that.

Reply:
Done.

Reply:
@awschristou - Can you assist moving this PR forward?